saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala

```diff
@@ -435,4 +458,17 @@ object MultiModalSpRLDataModel extends DataModel {
         else
           phraseConceptToWord(x.getSegmentConcept))
   }
+
+  private def getSegmentConcepts(img: String, seg: Integer): String = {
+    var concept = ""
+    val segmentFiltered = segments.getAllInstances.filter( s => {
+      s.getAssociatedImageID().equals(img) &&
+      s.getSegmentId()==(seg)
+    })
+    segmentFiltered.foreach(i=> {
+      concept = i.getSegmentConcept();
+      concept
+    })
+    concept
+  }
```

Comment:
I am not sure if this is correct. What is your goal here? have you tested its value?